### PR TITLE
Improve registry-facade HTTP requests resilience

### LINKED ISSUE #11847: Improve getBlob resilience to remote disconnections
### Bug description

error logs show remote disconnections while pulling from registries `read tcp 10.20.8.196:44218->74.125.199.128:443: read: connection reset by peer ` creating a fatal issue opening workspaces.

### Steps to reproduce

Not sure :smile: 

### Workspace affected

n/a

### Expected behavior

1. No issues. 
2. Have a simple retry mechanism for timeouts and disconnections
3. Have a meaningful metric for timeouts and disconnects

### Example repository

_No response_

### Anything else?

image

### PR DESCRIPTION
**Pull request by aledbf**
## Related Issue(s)

Fixes #11847

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by csweichel** [comment]
How can we test the retry mechanism, i.e. if the image pull succeeds?

**Comment by aledbf** [comment]
> How can we test the retry mechanism, i.e. if the image pull succeeds?

@csweichel please check the test

**Review by sagor999 — APPROVED** [review]

**Review by mustard-mh — APPROVED** [review]

File: components/registry-facade/pkg/registry/http_client.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+func defaultOptions() httpOpts {
+	return httpOpts{
+		RetryMax: 5,
+		Logger:   log.Log,
+
+		RequestLogHook: func(logger retryablehttp.Logger, req *http.Request, attempt int) {
+			if attempt > 0 {
+				log.Warnf("%v %v request failed. Retry count: %v\n", req.Method, req.URL, attempt)
```
Comment (jenting) [inline components/registry-facade/pkg/registry/http_client.go:42]:
```suggestion
                                log.Warnf("%v %v request failed. Retry count: %v", req.Method, req.URL, attempt)
```

**Review by jenting — COMMENTED** [review]
It'd be good if we could add one test case that the retry maximum exceeds.

**Review by andrew-farries — APPROVED** [review]

**Review by csweichel — APPROVED** [review]

File: components/registry-facade/pkg/registry/http_client.go (hunk 1 above)
Comment (aledbf) [reply components/registry-facade/pkg/registry/http_client.go]: done

**Review by geropl — APPROVED** [review]
LGTM

**Review by MrSimonEmms — APPROVED** [review]
